sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/AbstractRulesDefinition.java

```diff
@@ -92,21 +91,13 @@ public void define(Context context) {
     RulesDefinitionXmlLoader loader = new RulesDefinitionXmlLoader();
     loader.load(repository, new InputStreamReader(getClass().getResourceAsStream(rulesXmlFilePath), StandardCharsets.UTF_8));
 
-    allRuleKeys = new LinkedHashSet<>();
     for (NewRule rule : repository.rules()) {
       updateMetadata(rule);
-      allRuleKeys.add(rule.key());
     }
 
-    allRuleKeys = Collections.unmodifiableSet(allRuleKeys);
     repository.done();
   }
 
-  public Set<String> allRuleKeys() {
```

Comment:
`define` wasn't always called before. Based on the recommendations from JulienH I have changed the code to user the `RuleFinder`.